Sources/Noora/Utilities/Terminal.swift

```diff
@@ -9,15 +9,24 @@ import Foundation
 public protocol Terminaling {
     var isInteractive: Bool { get }
     var isColored: Bool { get }
+    var size: TerminalSize? { get }
     func withoutCursor(_ body: () throws -> Void) rethrows
     func inRawMode(_ body: @escaping () throws -> Void) rethrows
     func readCharacter() -> Character?
+    func readCharacterNonBlocking() -> Character?
+}
+
+public struct TerminalSize {
+    let rows: Int
+    let columns: Int
 }
 
 public struct Terminal: Terminaling {
     public let isInteractive: Bool
     public let isColored: Bool
 
+    public var size: TerminalSize? { Terminal.size() }
```

Comment:
I'd turn this into a function to make it obvious that the size is obtained at runtime when the function is invoked.

Reply:
Perhaps you can turn this one from static to an instance function:

```swift
public func size() -> TerminalSize? {
```